frontend/public/components/utils/name-value-editor.jsx

```diff
@@ -113,3 +100,129 @@ NameValueEditor.defaultProps = {
+  }
+};
+
+const itemTarget = {
+  hover(props, monitor, component) {
+    const dragIndex = monitor.getItem().index;
+    const hoverIndex = props.index;
+
+    // Don't replace items with themselves or with other row groupings on the page
+    if (dragIndex === hoverIndex || monitor.getItem().rowSourceId !== props.rowSourceId) {
+      return;
+    }
+
+    // Determine rectangle on screen
+    const hoverBoundingRect = component.node.getBoundingClientRect();
+
+    // Get vertical middle
+    const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+    // Determine mouse position
+    const clientOffset = monitor.getClientOffset();
+
+    // Get pixels to the top
+    const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+    // Only perform the move when the mouse has crossed half of the items height
+    // When dragging downwards, only move when the cursor is below 50%
+    // When dragging upwards, only move when the cursor is above 50%
+
+    // Dragging downwards
+    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+      return;
+    }
+
+    // Dragging upwards
+    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+      return;
+    }
+
+    // Time to actually perform the action
+    props.onMove(dragIndex, hoverIndex);
+
+    // Note: we're mutating the monitor item here!
+    // Generally it's better to avoid mutations,
+    // but it's good here for the sake of performance
+    // to avoid expensive index searches.
+    monitor.getItem().index = hoverIndex;
+  }
+};
+
+const collectSourcePair = (connect, monitor) => {
+  return {
+    connectDragSource: connect.dragSource(),
+    connectDragPreview: connect.dragPreview(),
+    isDragging: monitor.isDragging()
+  };
+};
+
+const collectTargetPair = (connect) => {
+  return {
+    connectDropTarget: connect.dropTarget(),
+  };
+};
+
+const PairElement = DragSource(DRAGGABLE_TYPE.ROW, pairSource, collectSourcePair)(DropTarget(DRAGGABLE_TYPE.ROW, itemTarget, collectTargetPair)(class PairElement extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const {isDragging, connectDragSource, connectDragPreview, connectDropTarget, nameString, valueString, allowSorting, readOnly, index, onChange, onRemove, pair} = this.props;
+    const deleteButton = <span><i className="fa fa-minus-circle pairs-list__btn pairs-list__delete-icon" aria-hidden={true} onClick={() => onRemove(index)}></i><span className="sr-only">Delete</span></span>;
+
+    return connectDropTarget(
+      connectDragPreview(
+        <div className={classNames('row', isDragging ? 'pairs-list__row-dragging' : 'pairs-list__row')} ref={node => this.node = node}>
+          <div className="col-xs-5 pairs-list__field">
+            <input type="text" className="form-control" placeholder={nameString.toLowerCase()} value={pair[NameValueEditorPair.Name]} onChange={e => onChange(e, index, NameValueEditorPair.Name)} disabled={readOnly}/>
```

Comment:
Arrow functions in `onChange` handlers are generally not a good pattern. This will create new functions every time render is called. It's better to use a named function (here and below).

Reply:
I updated the handlers to be behavior specific and added named functions instead of arrow functions. Also had to add back the constructor for binding purposes.